### PR TITLE
CTAD (deduction guides) for View

### PR DESCRIPTION
**Pull request by nliber**
Refactoring #5319 into smaller P/Rs

### CODE REVIEW

File: core/unit_test/TestViewCtorCTADs.hpp [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+struct TestViewCtorCTADs {
+  using DataType                     = double;
+  static constexpr DataType* m       = nullptr;
+  static constexpr const DataType* c = nullptr;
+
+  static void test_viewctorcopymove() {
+    Kokkos::View<DataType> vt;
+    Kokkos::View cd(vt);
+    ASSERT_TRUE((std::is_same_v<decltype(cd), decltype(vt)>));
```
[the remark this answers is not in the document]
Comment (nliber) [reply core/unit_test/TestViewCtorCTADs.hpp:57]: All the CTAD tests can be compile-only.  Why this particular one?

File: core/src/Kokkos_View.hpp [hunk 2]
```diff
@@ -1708,6 +1708,36 @@ class View : public ViewTraits<DataType, Properties...> {
   }
 };
 
+// These deduction guides result in an unmanaged View
+template <typename T>
+View(T*)->View<T>;
```
Comment (crtrott) [inline core/src/Kokkos_View.hpp:1713]: Bringing up the question again: should this be AnonymousSpace?

**Review by crtrott — CHANGES_REQUESTED** [review]
Blocking for now until we come to a resolution which memory space should be deduced.

File: core/src/Kokkos_View.hpp (hunk 2 above)
Comment (dalg24) [reply core/src/Kokkos_View.hpp:1713]: Or better: Do nothing ie do not provide deduction guidelines at all

**Comment by dalg24** [comment]
Consensus to close

**Pull request closed without merging**